pypeit/scripts/identify.py

```diff
@@ -86,55 +96,240 @@ def main(args):
+                                print("Sorry, try that again...")
+                                #better try again... Return to the start of the loop
+                                continue
+                            try:
+                                max_wav_str = input('Maximum Wavelength = ')
+                                max_wav = float(max_wav_str)
+
+                            except ValueError:
+                                print("Sorry, try that again...")
+                                #better try again... Return to the start of the loop
+                                continue
+                            else:
+                                #age was successfully parsed!
+                                #we're ready to exit the loop.
+                                break
+                        nspec = np.shape(arccen)[0]
+                        wav_x = np.linspace(min_wav, max_wav, nspec)
+                        custom_wav.append(wav_x)
+                        custom_wav_ind.append(slit_val)
+                        # sample the new solution to make a fake IDs list:
+                        lines_pix_arr.append(np.linspace(0, nspec-1, 8)[1:-1])
+                        lines_wav_arr.append(np.linspace(min_wav, max_wav, 8)[1:-1])
+                        lines_fit_ord.append([1])
+
+            if not np.any(wv_calib.wv_fits):
+                wv_calib.wv_fits = np.array(wv_fits_arr)
+                #wv_calib.to_file()
+            if not args.new_sol:
+                wv_calib.wv_fit2d=None
+            if args.new_sol:
+                wv_calib.copy_calib_internals(msarc)
+
+            # TODO: Make the following more elegant:
+            # fill lines with dummy values to make this work
+            # Ask the user if they wish to store the result in PypeIt calibrations
+            arcfitter.store_solution_multi(final_fit, slits.binspec,
+                                    wvcalib=wv_calib,
+                                    rmstol=args.rmstol,
+                                    force_save=args.force_save, 
+                                    multi = True, fits_dicts = fits_dicts,
+                                    specdata = np.array(specdata),
+                                    slits = slits, 
+                                    lines_pix_arr = lines_pix_arr,
+                                    lines_wav_arr = lines_wav_arr,
+                                    lines_fit_ord = np.array(lines_fit_ord), 
+                                    custom_wav = np.array(custom_wav), 
+                                    custom_wav_ind = np.array(custom_wav_ind) )
+            
+
+        # If we just want the normal one-trace output
         else:
-            waveCalib = None
+            # Load the calibration frame (if it exists and is desired).  Bad-pixel mask
+            # set to any flagged pixel in Arc.
+            wavecal = BuildWaveCalib(msarc, slits, spec, par, lamps, det=args.det,
```

Comment:
I think the exact same code appears in the `args.multi=True` line. So, I'd encourage you to reduce the replicated code, to make it easier to manage the code long term.

Reply:
I've consolidated the code.